include/internal/refcount.h

```diff
@@ -112,9 +113,12 @@ static __inline int CRYPTO_DOWN_REF(volatile int *val, int *ret, void *lock)
     *ret = _InterlockedExchangeAdd(val, -1) - 1;
     return 1;
 }
+#   endif
+
 #  endif
+# endif
 
-# else
+# ifndef HAVE_ATOMICS
```

Comment:
The most tricky part was to understand this change here, because it requires looking at all the entire ifdef-spagetti. But I think now that it's ok.

Reply:
I've added a bit of commentary, I hope that helps

Reply:
Yup!

Reply:
BTW and FYI, just in case you ever had the same problem:

I found it hard to follow the preprocessor indentation using Emacs' incremental search, because by default it treats a single literal space character as 'any whitespace'. But I found a way to change that:

https://emacs.stackexchange.com/questions/2819/match-two-spaces-with-incremental-search